File: guides/micronaut-patterns-composite/java/src/test/java/example/micronaut/ColorFetcherTest.java
package example.micronaut;

import io.micronaut.context.BeanContext;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import jakarta.inject.Inject;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertTrue;

@MicronautTest(startApplication = false)
public class ColorFetcherTest {

    @Inject
    BeanContext beanContext;

    @Test
    void beanOfTypeColorFetcherExists() {
        assertTrue(beanContext.containsBean(ColorFetcher.class));
    }
}

File: guides/micronaut-patterns-composite/kotlin/src/test/kotlin/example/micronaut/ColorFetcherTest.kt
package example.micronaut

import io.micronaut.context.BeanContext
import io.micronaut.test.extensions.junit5.annotation.MicronautTest
import jakarta.inject.Inject
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test

@MicronautTest(startApplication = false)
class ColorFetcherTest {

    @Inject
    lateinit var beanContext: BeanContext

    @Test
    fun beanOfTypeColorFetcherExists() {
        assertTrue(beanContext.containsBean(ColorFetcher::class.java))
    }
}
